templates/commands/render.go

```diff
@@ -322,6 +328,35 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// validateAndDefaultInputs validates and sets the input values provided by the user.
+func validateAndDefaultInputs(flagInputs map[string]string, spec *model.Spec) error {
+	var result *multierror.Error
+
+	for _, i := range spec.Inputs {
+		val, ok := flagInputs[i.Name.Val]
+		if !ok || val == "" {
+			if i.Default.Val != "" {
+				flagInputs[i.Name.Val] = i.Default.Val
+			} else if i.Required.Val {
+				result = multierror.Append(result, fmt.Errorf("%s", i.Name.Val))
+			}
+		}
+	}
+
+	if result != nil {
+		result.ErrorFormat = func(errors []error) string {
+			out := strings.Builder{}
+
+			for _, e := range errors {
+				out.WriteString(fmt.Sprintf("\n  - %s", e))
```

Comment:
Maybe we should print full help text here: input name, description, and default. The template might not be documented and they may not know how to look at the spec file, so this may be the only way the user finds out what inputs are needed.

I guess we should also build a `abc templates describe <url>` command so people can find out about inputs, I'll file an issue for that.

Reply:
Torn here, we dont want to overwhelm the user with a large amount of output. Maybe after we implement describe we can inform the user how to describe the template so they can learn the inputs from there.

Reply:
Created https://github.com/abcxyz/abc/issues/59 to track, we can resolve later.